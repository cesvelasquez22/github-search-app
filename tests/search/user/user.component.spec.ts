import { ComponentFixture, TestBed } from '@angular/core/testing';

import { UserComponent } from '../../../src/app/search/user/user.component';

describe('UserComponent', () => {
  let component: UserComponent;
  let fixture: ComponentFixture<UserComponent>;
  let compiled: HTMLElement

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ UserComponent ]
    })
    .compileComponents();

    fixture = TestBed.createComponent(UserComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });

  it('should render user', () => {
    component.user = {
      login: 'test',
      id: 1,
      avatar_url: 'test',
      html_url: 'test',
      score: 1,
    };
    fixture.detectChanges();
    expect(compiled).toMatchSnapshot();
  });

  it('should open user on _blank', () => {
    component.user = {
      login: 'test',
      id: 1,
      avatar_url: 'test',
      html_url: 'test',
      score: 1,
    };
    fixture.detectChanges();
    const a = compiled.querySelector('a');
    expect(a?.target).toBe('_blank');
  });
});
