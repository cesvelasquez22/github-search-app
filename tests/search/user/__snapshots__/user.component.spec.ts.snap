// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <a
    class="user"
    href=""
    target="_blank"
  >
    <div
      class="user-info"
    >
      <img
        alt="user_avatar"
        src=""
      />
    </div>
    <h3
      class="user-name truncate"
    >
      
    </h3>
    <p
      class="user-type"
    >
      
    </p>
  </a>
</div>
`;

exports[`UserComponent should render user 1`] = `
<div
  id="root2"
>
  <a
    class="user"
    href="test"
    target="_blank"
  >
    <div
      class="user-info"
    >
      <img
        alt="user_avatar"
        src="test"
      />
    </div>
    <h3
      class="user-name truncate"
    >
      test
    </h3>
    <p
      class="user-type"
    >
      
    </p>
  </a>
</div>
`;
